subsys/bluetooth/host/hci_core.c

```diff
@@ -4020,7 +4020,7 @@ void bt_data_parse(struct net_buf_simple *ad,
 		}
 
 		if (len > ad->len) {
-			BT_WARN("Malformed data");
+			BT_WARN("%s: malformed advertising data", __func__);
```

Comment:
Why the addition of the function name? 

If a user wants the function name in the warning logs, they can just enable `CONFIG_LOG_FUNC_NAME_PREFIX_WRN`

Reply:
well, maybe that symbol should be selected by `CONFIG_BT_DEBUG_LOG`. I can make another commit to do so if you agree.
Having app developers needing to grep for log messages is not very nice IMO.

Reply:
> well, maybe that symbol should be selected by `CONFIG_BT_DEBUG_LOG`. I can make another commit to do so if you agree. Having app developers needing to grep for log messages is not very nice IMO.

Yeah, honestly I do not know why `CONFIG_LOG_FUNC_NAME_PREFIX_WRN` and `CONFIG_LOG_FUNC_NAME_PREFIX_ERR` for that matter is not selected when debugging. In many cases the name of the function for errors and warnings are just as, or even more, useful than the names of the functions for debug log. 

I wouldn't mind if those were selected when doing debug logs :)